thorlcr/master/thgraphmanager.cpp

```diff
@@ -1358,6 +1343,50 @@ void publishPodNames(IWorkUnit *workunit, const char *graphName)
     }
 }
 
+static void auditThorSystemEventBuilder(std::string &msg, const char *eventName, std::initializer_list<const char*> args)
+{
+    msg += std::string(",Progress,Thor,") + eventName + "," + getComponentConfigSP()->queryProp("@name");
+    for (auto arg : args)
+        msg += "," + std::string(arg);
+    if (isContainerized())
+        msg += std::string(",") + k8s::queryMyPodName() + "," + k8s::queryMyContainerName();
+    else
+    {
+        const char *nodeGroup = queryServerStatus().queryProperties()->queryProp("@nodeGroup");
+        const char *queueName = queryServerStatus().queryProperties()->queryProp("@queue");
+        const char *logUrl = queryServerStatus().queryProperties()->queryProp("LogFile");
+        msg += std::string(",") + nodeGroup + "," + queueName + "," + logUrl;
+    }
+}
+
+void auditThorSystemEvent(const char *eventName)
+{
+    std::string msg;
+    auditThorSystemEventBuilder(msg, eventName, {});
+    LOG(MCauditInfo, "%s", msg.c_str());
+}
+
+void auditThorSystemEvent(const char *eventName, std::initializer_list<const char*> args)
+{
+    std::string msg;
+    auditThorSystemEventBuilder(msg, eventName, args);
+    LOG(MCauditInfo, "%s", msg.c_str());
+}
+
+void auditThorJobEvent(const char *eventName, const char *wuid, const char *graphName, const char *user)
+{
+    std::string msg = std::string(",Progress,Thor,") + eventName + "," + getComponentConfigSP()->queryProp("@name") + "," + wuid + "," + graphName + "," + nullText(user);
```

Comment:
Could you use

auditThorSystemEventBuilder(msg, { graphName, nullText(user) });

?

Reply:
yes, should have done, will change